test/unit_test/utils_test/test_register_custom_op.py

```diff
@@ -0,0 +1,441 @@
+        self.assertEqual(list(result.shape), expected_shape)
+
+    def test_circle_transpose_conv_basic(self):
+        """Test CircleTransposeConv basic functionality"""
+        input_tensor = torch.randn(1, 16, 16, 16)  # NHWC
+        weight_tensor = torch.randn(16, 3, 3, 16)  # OHWI format
+
+        result = torch.ops.circle_custom.transpose_conv(input_tensor, weight_tensor)
+
+        # Check output shape
+        expected_shape = [1, 18, 18, 16]  # NHWC with no padding
+        self.assertEqual(list(result.shape), expected_shape)
+
+    def test_circle_transpose_conv_with_custom_params(self):
+        """Test CircleTransposeConv with custom parameters"""
+        input_tensor = torch.randn(1, 16, 16, 16)
+        weight_tensor = torch.randn(16, 3, 3, 16)  # Output channels is 16
+        bias_tensor = torch.randn(16)  # Bias size should match output channels
+        stride = [2, 2]
+        padding = [1, 1]
+        output_padding = [1, 1]
+
+        result = torch.ops.circle_custom.transpose_conv(
+            input_tensor, weight_tensor, bias_tensor, stride, padding, output_padding
+        )
+
+        # Check output shape with custom params
+        expected_shape = [
+            1,
+            32,
+            32,
+            16,
+        ]  # NHWC with stride=2, padding=1, output_padding=1
+        self.assertEqual(list(result.shape), expected_shape)
+
+    def test_circle_transpose_conv_invalid_groups(self):
+        """Test CircleTransposeConv with invalid groups parameter"""
+        input_tensor = torch.randn(1, 16, 16, 16)
+        weight_tensor = torch.randn(16, 3, 3, 16)
+
+        with self.assertRaises(RuntimeError) as context:
+            torch.ops.circle_custom.transpose_conv(
+                input_tensor, weight_tensor, groups=2
+            )
+
+        self.assertIn(
+            "CircleTransposeConv only supports 1 'groups'", str(context.exception)
+        )
+
+    def test_circle_maxpool2d_with_defaults(self):
+        """Test CircleMaxPool2D with default parameters"""
+        input_tensor = torch.randn(1, 32, 32, 3)
+        kernel_size = [2, 2]
+
+        result = torch.ops.circle_custom.maxpool2d(input_tensor, kernel_size)
+
+        # Check output shape
+        expected_shape = [1, 16, 16, 3]  # NHWC with kernel_size=2, stride=2
+        self.assertEqual(list(result.shape), expected_shape)
+
+    def test_circle_maxpool2d_with_custom_params(self):
+        """Test CircleMaxPool2D with custom parameters"""
+        input_tensor = torch.randn(1, 32, 32, 3)
+        kernel_size = [3, 3]
+        stride = [1, 1]
+        padding = [1, 1]
+        dilation = [1, 1]
+        ceil_mode = True
+
+        result = torch.ops.circle_custom.maxpool2d(
+            input_tensor, kernel_size, stride, padding, dilation, ceil_mode
+        )
+
+        # Check output shape with custom params
+        expected_shape = [1, 32, 32, 3]  # NHWC with kernel_size=3, stride=1, padding=1
```

Comment:
```suggestion
        expected_shape = [1, 32, 32, 3]  # NHWC with kernel_size=3, stride=1, padding=1, dilation=1
```
(optional)
It is not that critical, but there is no description for `dilation`

Honestly, I think that these kinds of comments are not so informative, so it has no problem to be removed 🤔
```suggestion
        expected_shape = [1, 32, 32, 3]
```